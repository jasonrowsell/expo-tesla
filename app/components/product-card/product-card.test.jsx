import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { Alert } from 'react-native';

import ProductCard from './product-card';

describe('<CarCard />', () => {
  const renderComponent = () => render(
    <ProductCard title="title" tagline="tagline" source="img" />,
  );

  test('has 3 children', () => {
    const tree = renderer.create(
      <ProductCard title="title" tagline="tagline" source="img" />,
    ).toJSON();
    expect(tree.children.length).toBe(3);
  });

  test('renders correctly', () => {
    const tree = renderComponent();
    expect(tree).toMatchSnapshot();
  });

  test('has the title', () => {
    const { getByText } = renderComponent();
    getByText('title');
  });

  test('Order now alert is displayed when order now button is pressed', () => {
    jest.spyOn(Alert, 'alert');
    const { getByText } = renderComponent();

    const button = getByText('Order Now');
    fireEvent.press(button);

    expect(Alert.alert).toHaveBeenCalledWith('Order Now was pressed');
  });

  test('Learn more alert is displayed when learn more button is pressed', () => {
    jest.spyOn(Alert, 'alert');
    const { getByText } = renderComponent();

    const button = getByText('Learn More');
    fireEvent.press(button);

    expect(Alert.alert).toHaveBeenCalledWith('Learn More was pressed');
  });

  describe(('termsLinks enabled'), () => {
    test('renders terms links', () => {
      const { getByText } = render(
        <ProductCard title="title" tagline="tagline" source="img" termsLinks />,
      );

      getByText('Jason Rowsell Â© 2021');
      getByText('Privacy & Legal');
      getByText('Careers');
      getByText('News');
    });

    test('Shop now alert is displayed when shop now button is pressed', () => {
      jest.spyOn(Alert, 'alert');
      const { getByText } = render(
        <ProductCard title="title" tagline="tagline" source="img" termsLinks />,
      );

      const button = getByText('Shop Now');
      fireEvent.press(button);

      expect(Alert.alert).toHaveBeenCalledWith('Shop Now was pressed');
    });
  });
});
