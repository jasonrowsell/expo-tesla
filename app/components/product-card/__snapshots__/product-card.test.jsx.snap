// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarCard /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "position": "absolute",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={null}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": "30%",
        "opacity": undefined,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "display": "flex",
          "fontSize": 35,
          "fontWeight": "500",
          "paddingBottom": 5,
        }
      }
    >
      title
    </Text>
    <Text
      style={
        Object {
          "color": "#5c5e62",
          "fontSize": 17,
        }
      }
    >
      tagline
    </Text>
  </View>
  <View
    style={
      Object {
        "bottom": 80,
        "opacity": undefined,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#171A20CC",
            },
          ]
        }
        testID="button"
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          buttonContent
        </Text>
      </View>
    </View>
  </View>
</View>
`;
